### ISSUE TITLE
Cifar10DataSetIterator exception on next() in MobileNetTransferLearningExample

### ISSUE THREAD

**Issue by MrForum** [deeplearning4j/deeplearning4j#9318]
I get the below exception from Cifar10DataSetIterator.next() when running MobileNetTransferLearningExample, or the following simplified standalone code:

Code: 
import org.deeplearning4j.datasets.iterator.impl.Cifar10DataSetIterator;
import org.nd4j.linalg.dataset.DataSet;
public class CFarTest {
	public static void main(String args[]) {
		try {
			Cifar10DataSetIterator it = new Cifar10DataSetIterator(10);
			DataSet ds = it.next();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

Output: 
java.lang.UnsupportedOperationException: Cannot do conversion to one hot using batched reader: 10 output classes, but array.size(1) is 3 (must be equal to 1 or numClasses = 10)
	at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertWritablesBatched(RecordReaderMultiDataSetIterator.java:414)
	at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertFeaturesOrLabels(RecordReaderMultiDataSetIterator.java:360)
	at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.nextMultiDataSet(RecordReaderMultiDataSetIterator.java:333)
	at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.next(RecordReaderMultiDataSetIterator.java:213)
	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:378)
	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:453)
	at mobilenet.CFarTest.main(CFarTest.java:10)

**Comment by MrForum** [comment]
Output from MobileNetTransferLearningExample below:


base) [forum@localhost target]$ pwd
/home/forum/projects/deeplearning4j-examples/tensorflow-keras-import-examples/target
(base) [forum@localhost target]$ java -cp .:samediff-examples-1.0.0-beta7-shaded.jar org.deeplearning4j.modelimportexamples.tf.advanced.mobilenet.MobileNetTransferLearningExample
...
16:45:00.174 [main] INFO org.datavec.image.recordreader.BaseImageRecordReader - ImageRecordReader: 3 label classes inferred using label generator ParentPathLabelGenerator
Exception in thread "main" java.lang.UnsupportedOperationException: Cannot do conversion to one hot using batched reader: 10 output classes, but array.size(1) is 3 (must be equal to 1 or numClasses = 10)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertWritablesBatched(RecordReaderMultiDataSetIterator.java:414)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertFeaturesOrLabels(RecordReaderMultiDataSetIterator.java:360)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.nextMultiDataSet(RecordReaderMultiDataSetIterator.java:333)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.next(RecordReaderMultiDataSetIterator.java:213)
        at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:378)
        at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:453)
        at org.deeplearning4j.modelimportexamples.tf.advanced.mobilenet.MobileNetTransferLearningExample.main(MobileNetTransferLearningExample.java:223)

**Comment by MrForum** [comment]
The same issue appears to break the CIFARClassifier example:

(base) [forum@localhost target]$ java -jar ./dl4j-examples-1.0.0-beta7-shaded.jar org.deeplearning4j.examples.quickstart.modeling.convolution.CIFARClassifier
no main manifest attribute, in ./dl4j-examples-1.0.0-beta7-shaded.jar
(base) [forum@localhost target]$ java -cp .:dl4j-examples-1.0.0-beta7-shaded.jar org.deeplearning4j.examples.quickstart.modeling.convolution.CIFARClassifier
o.d.i.r.BaseImageRecordReader - ImageRecordReader: 3 label classes inferred using label generator ParentPathLabelGenerator
o.d.i.r.BaseImageRecordReader - ImageRecordReader: 10 label classes inferred using label generator ParentPathLabelGenerator
o.d.e.q.m.c.CIFARClassifier - Building simple convolutional network...
o.n.l.f.Nd4jBackend - Loaded [CpuBackend] backend
o.n.n.NativeOpsHolder - Number of threads used for linear algebra: 1
o.n.l.c.n.CpuNDArrayFactory - *********************************** CPU Feature Check Warning ***********************************
o.n.l.c.n.CpuNDArrayFactory - Warning: Initializing ND4J with Generic x86 binary on a CPU with AVX/AVX2 support
o.n.l.c.n.CpuNDArrayFactory - Using ND4J with AVX/AVX2 will improve performance. See deeplearning4j.org/cpu for more details
o.n.l.c.n.CpuNDArrayFactory - Or set environment variable ND4J_IGNORE_AVX=true to suppress this warning
o.n.l.c.n.CpuNDArrayFactory - *************************************************************************************************
o.n.n.Nd4jBlas - Number of threads used for OpenMP BLAS: 1
o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CPU]; OS: [Linux]
o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [10.6GB];
o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [OPENBLAS]
o.d.n.m.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE]
o.d.u.VertxUIServer - Deeplearning4j UI server started at: http://localhost:9000
o.d.u.VertxUIServer - StatsStorage instance attached to UI: FileStatsStorage(/tmp/ui-stats.dl4j)
Exception in thread "ADSI prefetch thread" java.lang.RuntimeException: java.lang.UnsupportedOperationException: Cannot do conversion to one hot using batched reader: 10 output classes, but array.size(1) is 3 (must be equal to 1 or numClasses = 10)
        at org.nd4j.linalg.dataset.AsyncDataSetIterator$AsyncPrefetchThread.run(AsyncDataSetIterator.java:437)
Caused by: java.lang.UnsupportedOperationException: Cannot do conversion to one hot using batched reader: 10 output classes, but array.size(1) is 3 (must be equal to 1 or numClasses = 10)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertWritablesBatched(RecordReaderMultiDataSetIterator.java:414)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.convertFeaturesOrLabels(RecordReaderMultiDataSetIterator.java:360)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.nextMultiDataSet(RecordReaderMultiDataSetIterator.java:333)
        at org.deeplearning4j.datasets.datavec.RecordReaderMultiDataSetIterator.next(RecordReaderMultiDataSetIterator.java:213)
        at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:378)
        at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:453)
        at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:85)
        at org.nd4j.linalg.dataset.AsyncDataSetIterator$AsyncPrefetchThread.run(AsyncDataSetIterator.java:411)

**Comment by agibsonccc** [comment]
@MrForum are you just running the examples out of the box? Sorry a release is coming out in the next few days and then I'll take a look at this if it's actually a bug. @treo do you remember something like this when you were doing the examples audit?

**Comment by MrForum** [comment]
@agibsonccc yes this are examples unmodified run from master pulled today.  Thanks.

**Comment by treo** [comment]
I just re-ran that example and it works just fine for me. 

The error message looks like you either are using different data (with 3 classes instead of the 10 that cifar10 has) or your data download of cifar10 is corrupted.

**Comment by MrForum** [comment]
@treo I just re-cloned, rebuilt, and re-ran using above commands (no code modification) and had the same error.  Can you tell me where the data is downloaded so I can check for corruption, or give me the list of commands that should work on a fresh pull?  Thank you.

**Comment by treo** [comment]
The data should be in `~/.deeplearning4j/data/cifar10`. You should find a labels.txt file in there, as well as two directories: test and train, and both of these directories should have 10 directories each, labeled 0 to 9.

**Comment by MrForum** [comment]
Looks like 10 folders in test, but only 3 in train.  How would that have happened?  If I delete cifar10 will it redownload?  Thanks. 


(base) [forum@localhost .deeplearning4j]$ cd data/cifar10/
(base) [forum@localhost cifar10]$ ll
total 8
-rw-rw-r--.  1 forum forum  60 May 20 15:35 labels.txt
-rw-rw-r--.  1 forum forum 559 May 20 15:35 sort.js
drwxrwxr-x. 12 forum forum  96 May 20 15:35 test
drwxrwxr-x.  5 forum forum  33 May 20 15:35 train
(base) [forum@localhost cifar10]$ ll test/
total 496
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 0
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 1
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 2
drwxrwxr-x. 2 forum forum 24576 May 20 15:35 3
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 4
drwxrwxr-x. 2 forum forum 24576 May 20 15:35 5
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 6
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 7
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 8
drwxrwxr-x. 2 forum forum 32768 May 20 15:35 9
(base) [forum@localhost cifar10]$ ll train/
total 684
drwxrwxr-x. 2 forum forum 155648 May 20 15:35 5
drwxrwxr-x. 2 forum forum 151552 May 20 15:35 8
drwxrwxr-x. 2 forum forum 163840 May 20 15:35 9
(base) [forum@localhost cifar10]$

**Comment by treo** [comment]
There are multiple ways how that might happen, the most likely is that you managed to stop it at the exact point while it was extracting it for the first time. 

When you delete that folder, it should re-download and unpack it again.

**Comment by MrForum** [comment]
Deleting the folder fixed it, thank you.
